### PR TITLE
Feat/v2

### PR DESCRIPTION
**Pull request by Dougley**
This PR brings a full Typescript rewrite of all components

## What's different?
- Tyr is no longer dependent on Eris . In the past we depended on Eris' methods for sending Websocket events
  - Websocket event are now abstracted, meaning they can be implemented in any library
- A function in the Player class has been renamed
  - `setVolume()` > `volume()`
- `Player.play` now accepts Lavalink's starting modifiers such as `startTime`, `noReplace`
- `LavalinkVoiceConnectionManager` has been renamed to `ErisPlayerManager` and is no longer exported under the module index, import `@thesharks/tyr/eris` for access to `ErisPlayerManager`

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
   parserOptions: {
-    ecmaVersion: 2018
+    ecmaVersion: 12,
+    sourceType: 'module'
   },
+  plugins: [
+    '@typescript-eslint'
+  ],
   rules: {
+    'no-undef': 'off'
```
Comment (linuswillner) [inline .eslintrc.js:18]: Why are we turning off the no-undef rule? Seems like a pretty big omission 🤔

File: src/components/ErisPlayerManager.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+      }
+      this.delete(data.guild_id)
+    })
+    player.connect({
+      sessionId: data.session_id,
+      event: data
+    })
+  }
+
+  private selectBestLavalinkNode (endpoint: string) {
```
Comment (linuswillner) [inline src/components/ErisPlayerManager.ts:98]: Might want to document what "best" entails here, meaning how the algorithm works. From the looks of it, this seems to be by lowest system load in the current region.

**Review by linuswillner — APPROVED** [review]
Just a few nits here, but those are by no means any showstoppers. Good work.

File: .eslintrc.js (hunk 1 above)
Comment (Dougley) [reply .eslintrc.js:18]: TSC catches undef errors, eslint has issues with things like interfaces being "undefined"

**Pull request merged**